### PR TITLE
[Fix] Fix errors about deploying MMYOLO-OpenVINO, DETR, ConvFormer and RTMDet

### PR DESCRIPTION
**Pull request by hanrui1sensetime**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Fix bugs.

## Modification

- [x] MMYolo OpenVINO configs in regression test.
- [x] DETR after refactor in mmdet 3.0.0rc6 support(DetectionTransformer detector) deploy support.
- [x] RTMDet SDK precision loss fix.
- [x] ConvFormer score loss fix and SDK failed fix(sdk result is slightly different from pytorch inference because the resize using pillow and mmcv are different).
- [x] Related UT above.

## BC-breaking (Optional)

None.

## Use cases (Optional)

None.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: tests/regression/mmdet.yml [hunk 1]
```diff
@@ -250,7 +250,9 @@ models:
       - *pipeline_ort_dynamic_fp32
       - *pipeline_trt_dynamic_fp32
       - *pipeline_ncnn_static_fp32
-      - *pipeline_openvino_dynamic_fp32
+      - deploy_config: configs/mmdet/detection/detection_openvino_dynamic-640x640.py
+        convert_image: *convert_image
+        backend_test: False
```
[the remark this answers is not in the document]
Comment (hanrui1sensetime) [reply tests/regression/mmdet.yml:255]: > why `backend_test` is false?

Generally, we only test tensorrt sdk on regression test.

File: csrc/mmdeploy/codebase/mmcls/conformer_cls.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    return to_value(std::move(output));
+  }
+
+ private:
+  static constexpr const auto kHost = Device{0};
+
+  int topk_{1};
+};
+
+MMDEPLOY_REGISTER_CODEBASE_COMPONENT(MMClassification, ConformerHead);
```
Comment (lvhan028) [inline csrc/mmdeploy/codebase/mmcls/conformer_cls.cpp:60]: It seems `class LinearClsHead` and `class ConformerHead` has no difference, right?
You can try
```
using ConformerHead = LinearClsHead;
MMDEPLOY_REGISTER_CODEBASE_COMPONENT(MMClassification, ConformerHead);
```
, just like what we did for mmpose. You can refer to https://github.com/open-mmlab/mmdeploy/blob/ea73f20fe43e912d25d25985a4c0320fca3c6a4f/csrc/mmdeploy/codebase/mmpose/keypoints_from_heatmap.cpp#L362

File: mmdeploy/codebase/mmcls/models/backbones/conformer.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+from mmdeploy.utils import Backend
+
+
+@FUNCTION_REWRITER.register_rewriter(
+    func_name='mmcls.models.backbones.conformer.FCUUp.forward',
+    backend=Backend.TENSORRT.value)
+def fcuup__forward__tensorrt(self, x, H, W):
+    """Rewrite `forward` of FCUUp used in conformer.
+
+    FCUUp uses nearest interpolate while trt interpolate default
```
Comment (grimoire) [inline mmdeploy/codebase/mmcls/models/backbones/conformer.py:14]: If this is true, can we rewrite `F.interpolate`?

File: mmdeploy/codebase/mmdet/models/dense_heads/detr_head.py [hunk 4]
```diff
@@ -9,28 +9,17 @@
 
 
 @FUNCTION_REWRITER.register_rewriter(
-    'mmdet.models.dense_heads.DETRHead.forward_single')
-def detrhead__forward_single__default(self, x, img_metas):
-    """forward_single of DETRHead.
+    'mmdet.models.dense_heads.DETRHead.forward')
```
Comment (grimoire) [inline mmdeploy/codebase/mmdet/models/dense_heads/detr_head.py:12]: What is the difference from the origin forward?

**Review by grimoire — APPROVED** [review]

**Review by lvhan028 — APPROVED** [review]

**Review by RunningLeon — COMMENTED** [review]
regression test failed with detr+trt on torch1.8.0 1.10.0 and 1.12.0

**Comment by hanrui1sensetime** [comment]
> regression test failed with detr+trt on torch1.8.0 1.10.0 and 1.12.0

See https://github.com/NVIDIA/TensorRT/issues/2326

**Review by RunningLeon — APPROVED** [review]
LGTM
